CMakeLists.txt

```diff
@@ -50,7 +50,17 @@ if(APPLE)
       message(FATAL_ERROR "Unrecognized iOS platform '${IOS_PLAT}'")
     endif()
 
-    if(${IOS_ARCH} STREQUAL "x86_64")
+    if((${IOS_ARCH} STREQUAL "arm64") AND (${IOS_PLAT} STREQUAL "iphonesimulator"))
+      set(IOS_PLATFORM "iphonesimulator")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -arch arm64")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -arch arm64")
+      set(CMAKE_SYSTEM_PROCESSOR arm64)
+    elseif((${IOS_ARCH} STREQUAL "arm64e") AND (${IOS_PLAT} STREQUAL "iphonesimulator"))
+      set(IOS_PLATFORM "iphonesimulator")
+       set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -arch arm64e")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -arch arm64e")
+      set(CMAKE_SYSTEM_PROCESSOR arm64e)  
```

Comment:
Instead of adding lines from 53 to 62, can we just add ${IOS_PLAT} at lines 67 and 72;
```cmake
    elseif(${IOS_ARCH} STREQUAL "arm64")
          set(IOS_PLATFORM ${IOS_PLAT})
           ...
    elseif(${IOS_ARCH} STREQUAL "arm64e")
           set(IOS_PLATFORM ${IOS_PLAT})
            ...
    else()
```